lua/entities/gmod_wire_expression2/core/player.lua

```diff
@@ -533,3 +536,69 @@ e2function ranger entity:eyeTraceCursor()
 	ret.RealStartPos = this:GetShootPos()
 	return ret
 end
+
+--[[--------------------------------------------------------------------------------------------]]--
+
+local spawnAlert = {}
+local runBySpawn = 0
+local lastJoined = nil
+
+local leaveAlert = {}
+local runByLeave = 0
+local lastLeft = nil
```

Comment:
This needs to be set to NULL as if a player hasn't disconnected yet, `lastLeft` will still be nil. I'd change `lastJoined` to just to make it look consistent.